src/main/java/org/wildfly/security/sasl/util/SaslMechanismInformation.java

```diff
@@ -679,6 +680,43 @@ public static boolean needsServerCredentials(final String mechName) {
         }
     }
 
+    /**
+     * Determine whether the given mechanism name is known to not use any sort of {@link Principal} for authentication.
+     *
+     * @param mechName the mechanism name (must not be {@code null})
+     * @return {@code true} if the mechanism does not use a principal, {@code false} if it does or it is not known
+     */
+    public static boolean doesNotUsePrincipal(final String mechName) {
+        switch (mechName) {
+            case Names.ANONYMOUS:
```

Comment:
Not so sure we can say that 100% of the times, the mechanism itself does allow the user to send an authentication ID, it just doesn't do anything to authenticate them,